packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -392,6 +392,7 @@ export const scheduleTimeout: any =
 export const cancelTimeout: any =
   typeof clearTimeout === 'function' ? clearTimeout : (undefined: any);
 export const noTimeout = -1;
+const localPromise = Promise;
```

Comment:
This will error if it's not defined. You need to run the `typeof Promise` check directly and not through an indirection.

Reply:
E.g. `const localPromise = typeof Promise === 'function' ? Promise : undefined;

Reply:
🤦

Reply:
sharp eye